pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -343,6 +359,13 @@ func gatherDestinations(weights []*networking.DestinationWeight) []*networking.D
 }
 
 // TODO: move up to more general location so this can be re-used
+// TODO: should this try to use `istio_route.ConvertDestinationToCluster`?
 func destToClusterName(d *networking.Destination) string {
 	return model.BuildSubsetKey(model.TrafficDirectionOutbound, d.Subset, d.Host, &model.Port{Name: d.Port.GetName()})
 }
+
+// TODO: move up to pkg/proto
+func pprint(p proto.Message) string {
+	s, _ := (&jsonpb.Marshaler{}).MarshalToString(p)
```

Comment:
suggest 
s, err := (&jsonpb.Marshaler{}).MarshalToString(p)
if err != nil {
  return err.Error()
}

Reply:
Given this is for debugging, I didn't worry about that. In fact, I should probably just remove the method from the PR.